### DISCUSSION TITLE
Trouble reading image

### DISCUSSION THREAD

**Discussion by rmc868** [ImageMagick/ImageMagick#2042] [Help]
Hello!

I am new to using libraries.  I got Magick++ working and can build and run a project where I create a blank canvas and output its width/height.  However, when I try to read an image - either from my project directory or a URL - I get the following exception:

Exception thrown at 0x00007FFDFC02C369 (msvcr120.dll) in Test.exe: 0xC0000005: Access violation reading location 0x000000FEFB300000.

This is the code:
![image](https://user-images.githubusercontent.com/22719773/82515512-5d7f4980-9ae6-11ea-8446-d48520bfcbbe.png)

Any help would be much appreciated.  Thanks!

**Comment by snibgo** [comment]
I suspect a build problem, with incompatible libraries or something. Can you read other file formats, eg tiff, gif and png? If the problem is just jpeg, then perhaps your jpeg library is bad. Or perhaps your msvc file is bad. For a pre-built IM binary, I get:

f:\web\im>dir %IMG7%ms*
 Volume in drive C is Windows
 Volume Serial Number is 7CB7-18B6

 Directory of c:\im\ImageMagick-7.0.8-64

04/10/2013  23:58           660,128 msvcp120.dll
04/10/2013  23:58           963,232 msvcr120.dll

**Reply by rmc868** [reply]
Thanks for your reply.  I can't read other file formats either.

msvc files:
![image](https://user-images.githubusercontent.com/22719773/82565786-f941a180-9b48-11ea-981c-c652709267b6.png)


Also, I notice that when I try to open magick.exe from my IM directory it looks like a window starts to launch but then closes immediately.  Not sure if that has an relevance here.  imdisplay.exe opens correctly, though

**Reply by rmc868** [reply]
Okay so the program works if I switch from Debug configuration to Release configuration. The exception described above happens on Debug configuration (which was the default)
![image](https://user-images.githubusercontent.com/22719773/82573958-71619480-9b54-11ea-9ff4-204734470488.png)


As far as I can see (with limited knowledge), the relevant settings seem to be the same between Debug and Release for me, differences being:
Debug has incremental linking enabled, where Release does not
Debug has certain optimization settings disabled, where Release has them enabled

My settings for where to find libraries and includes are the same in both configurations

Any ideas about why it won't work in Debug configuration?

**Reply by snibgo** [reply]
I don't know how you build IM. Perhaps you use Microsoft Visual C. There have been many comments in the old forum that debug mode doesn't work.

**Reply by rmc868** [reply]
I just downloaded the pre-built binary.  Are you saying maybe that pre-built binary is not fully compatible with Microsoft Visual C/Visual Studio and that a possible solution is to try to build from source using Microsoft Visual C? or something along those lines?

I'll try to check the old forum as well, thanks!

**Reply by rmc868** [reply]
Another issue is that even though it will build and run in Release mode within Visual Studio, if I find the .exe in the Release folder of my project and try to open it there, it doesn't work.  It opens a terminal and immediately closes.

**Comment by snibgo** [comment]
Sorry, I don't use Visual Studio, and can't advise.

**Comment by snibgo** [comment]
Sorry, I don't use Visual Studio, and can't advise on that.

But I suppose you have built a console program, so I suggest you run your program in a console (from the command line), not by double-clicking on icons or whatever.

**Reply by rmc868** [reply]
Oh I mean I normally do.  I assumed that double clicking the .exe would just launch and run in a console.  I just tried it from Windows Explorer on a whim and thought it was weird that it didn't work.  I see now that doesn't seem to work in general.  

I did get a bit farther but am now stuck at this error:

![image](https://user-images.githubusercontent.com/22719773/82716957-36e91c00-9c68-11ea-9227-a00b7d604a82.png)

It will read the image and get the width/height/format of the image but will not display it when I call image.display()

This happened yesterday while using the pre-built binary.  So today I tried uninstalling and then building from source.  I followed the directions for building from source on Windows with Visual Studio here in the github.  I made sure that I had the specified requirements and Visual Studio dependencies and I am still getting the same result and error.

If you need any specific information to help me out, let me know and I'll provide it.  Thanks

**Reply by dlemstra** [reply]
`image.Display()` is not supported under Windows because it requires the `X11` delegate.

**Reply by snibgo** [reply]
Clicking on an EXE will start a console, run the program, and close the console. This works the same as if you create a console and run the program by typing its name BUT because the console closes immediately, you can't read any error messages.

A dlemstra says, IM display functions use X11. Microsoft Windows doesn't have X11. I have heard that it is possible to install X11 on Windows, but I never have.
